Tasks/Common/func-tools-common/functoolsutility.ts

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as util from 'util';
+import * as tl from "azure-pipelines-task-lib/task";
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+import * as path from "path";
+
+const funcToolName = 'func';
+
+function getExecutableExtension(): string {
+    if (os.type().match(/^Win/)) {
+        return '.exe';
+    }
+
+    return '';
+}
+
+function getDownloadUrl(version: string) {
+    switch (os.type()) {
```

Comment:
Add a constant which is the download base url: `https://github.com/Azure/azure-functions-core-tools/releases/download/`
You could also use it like: ``` `${baseDownloadUrl}/${version}/Azure.Functions.Cli.${os}-x64.${version}.zip` ```